src/hooks/useOrders.ts

```diff
@@ -1,34 +1,91 @@
 import { useEffect, useCallback, useRef } from 'react'
-// eslint-disable-next-line @typescript-eslint/camelcase
-import { unstable_batchedUpdates } from 'react-dom'
+import { unstable_batchedUpdates as batchedUpdates } from 'react-dom'
 
 import useGlobalState from './useGlobalState'
 import { overwriteOrders, updateOffset, updateOrders } from 'reducers-actions/orders'
 
 import { useWalletConnection } from './useWalletConnection'
 import useSafeState from './useSafeState'
-import { exchangeApi } from 'api'
-import { AuctionElement } from 'api/exchange/ExchangeApi'
+import { exchangeApi, web3 } from 'api'
+import { AuctionElement, PendingTxObj } from 'api/exchange/ExchangeApi'
 import { useCheckWhenTimeRemainingInBatch } from './useTimeRemainingInBatch'
+import { removePendingOrdersAction } from 'reducers-actions/pendingOrders'
+import { setStorageItem } from 'utils'
 
 interface Result {
   orders: AuctionElement[]
+  pendingOrders: PendingTxObj[]
   forceOrdersRefresh: () => void
   isLoading: boolean
 }
 
+export const GP_PENDING_ORDER_KEY = 'GP_ORDER_TX_HASHES'
 const REFRESH_WHEN_SECONDS_LEFT = 60 // 1min before batch done
-// solutions submitted at this point
+const emptyArray: PendingTxObj[] = []
 
 export function useOrders(): Result {
   const { userAddress, networkId, blockNumber } = useWalletConnection()
   const [
     {
       orders: { orders, offset },
+      pendingOrders,
     },
     dispatch,
   ] = useGlobalState()
 
+  // Handle Pending Orders
+  // pending orders are saved in global app state as well as local storage
+  // users can refresh page after creating orders and see them after reload thanks for local storage
+  // uses this effect hook to:
+  //   1. listen to block updates and filter recently mined transactions based on their hash
+  //   2. remove any orders that may have been re-sent at a higher gas price
+  const currentPendingOrders: PendingTxObj[] =
+    (networkId && userAddress && pendingOrders[networkId] && pendingOrders[networkId][userAddress]) || emptyArray
+  useEffect(() => {
+    // Don't fire if there are no pending orders...
+    if (networkId && userAddress && currentPendingOrders.length > 0) {
+      const managePendingOrders = async (): Promise<void> => {
```

Comment:
Concern 2: If we implement it like this, what happens if 10 parts of the UI uses `useOrders`, does it execute this logic 10 times?
@alfetopito @W3stside @Velenir

Reply:
It does
Where would you rather put it?
Or maybe use cache?

Reply:
Cache would work, although also we have the "Concern 1" above, I wonder if this is more central piece